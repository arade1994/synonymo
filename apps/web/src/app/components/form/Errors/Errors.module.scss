.error {
  color: #e53e3e;
  font-size: 0.875rem;
  margin-top: 0.5rem;
}
