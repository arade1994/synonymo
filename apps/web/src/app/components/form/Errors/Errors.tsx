import cx from "classnames";

import styles from "./Errors.module.scss";

interface Props {
  errors: string[];
  className?: string;
}

const Errors: React.FC<React.PropsWithChildren<Props>> = ({
  errors,
  className,
}) => {
  return (
    errors && (
      <div className={cx(styles.error, className)}>{errors.join(", ")}</div>
    )
  );
};

export default Errors;
