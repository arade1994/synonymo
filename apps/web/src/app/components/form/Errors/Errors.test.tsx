import { describe, expect, test } from "vitest";

import { render } from "@testing-library/react";

import Errors from "./Errors";

describe("Errors", () => {
  test("It should match a snapshot when rendered by default with no errors", () => {
    const { baseElement } = render(<Errors errors={[]} />);
    expect(baseElement).toMatchSnapshot();
  });

  test("It should match a snapshot when rendered by default with errors", () => {
    const { baseElement } = render(<Errors errors={["test"]} />);
    expect(baseElement).toMatchSnapshot();
  });
});
