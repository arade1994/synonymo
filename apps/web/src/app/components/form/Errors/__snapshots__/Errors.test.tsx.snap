// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Errors > It should match a snapshot when rendered by default with errors 1`] = `
<body>
  <div>
    <div
      class="_error_b9420b"
    >
      test
    </div>
  </div>
</body>
`;

exports[`Errors > It should match a snapshot when rendered by default with no errors 1`] = `
<body>
  <div>
    <div
      class="_error_b9420b"
    />
  </div>
</body>
`;
